src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1274 @@
+
+         do j = 1,nlev_soildecomp_standard
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               if (j==1) w_scalar(c,:) = 0._r8
+               psi = min(soilpsi(c,j),maxpsi)
+               ! decomp only if soilpsi is higher than minpsi
+               if (psi > minpsi) then
+                  w_scalar(c,1) = w_scalar(c,1) + (log(minpsi/psi)/log(minpsi/maxpsi))*fr(c,j)
+               end if
+            end do
+         end do
+
+         ! TODO May need a hook from MIMICS to ch4 code for this to work
+         ! Calculate ANOXIA
+         ! anoxia = .true. when (use_lch4)
+
+         if (anoxia) then
+
+            do j = 1,nlev_soildecomp_standard
+               do fc = 1,num_soilc
+                  c = filter_soilc(fc)
+
+                  if (j==1) o_scalar(c,:) = 0._r8
+
+                  o_scalar(c,1) = o_scalar(c,1) + fr(c,j) * max(o2stress_unsat(c,j), mino2lim)
+               end do
+            end do
+         else
+            o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+         end if
+
+         deallocate(fr)
+
+      else
+
+         ! calculate the rate constant scalar for soil water content.
+         ! Uses the log relationship with water potential given in
+         ! Andren, O., and K. Paustian, 1987. Barley straw decomposition in the field:
+         ! a comparison of models. Ecology, 68(5):1190-1200.
+         ! and supported by data in
+         ! Orchard, V.A., and F.J. Cook, 1983. Relationship between soil respiration
+         ! and soil moisture. Soil Biol. Biochem., 15(4):447-453.
+
+         do j = 1,nlevdecomp
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               psi = min(soilpsi(c,j),maxpsi)
+               ! decomp only if soilpsi is higher than minpsi
+               if (psi > minpsi) then
+                  w_scalar(c,j) = (log(minpsi/psi)/log(minpsi/maxpsi))
+               else
+                  w_scalar(c,j) = 0._r8
+               end if
+            end do
+         end do
+
+         ! Calculate ANOXIA
+         ! anoxia = .true. when (use_lch4)
+
+         if (anoxia) then
+            do j = 1,nlevdecomp
+               do fc = 1,num_soilc
+                  c = filter_soilc(fc)
+
+                  o_scalar(c,j) = max(o2stress_unsat(c,j), mino2lim)
+               end do
+            end do
+         else
+            o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+         end if
+
+      end if
+
+      ! Term that reduces decomposition rate at depth
+      ! Placeholder. For now depth_scalar = 1.
+      do j = 1, nlevdecomp
+         do fc = 1, num_soilc
+            c = filter_soilc(fc)
+            ! Using fixed e-folding depth as in
+            ! SoilBiogeochemDecompCascadeBGCMod.F90
+!           depth_scalar(c,j) = exp(-zsoi(j) / decomp_depth_efolding)
+            depth_scalar(c,j) = 1.0_r8
+         end do
+      end do
+
+      mimics_fmet_p1 = params_inst%mimics_fmet(1)
```

Comment:
This makes me wonder if some of these shouldn't be arrays?

Reply:
If we prefer them as arrays, would you approve using the params_inst versions of them, which are already declared as arrays?

Reply:
I added a TODO directly in the code with Erik's suggestion to prevent forgetting it.